#150627360 Fix Front Tests

WinstonKamau opened the pull request:
## **What does this pull request do ?**
Acts on the specifications of completing tests that had not been implemented and adding asserts on the front tests folder.
## **Description of tasks to be completed?**
- Add a test that asserts team access works while adding channels.
- Add a test that asserts team access works while adding checks.
- Add a test that asserts bad kinds of channels are not added.
- Add a test that asserts that pushover requests are validated for their priority parameters.
- Add a test that asserts non-interference on team access.

## **How should this be manually tested?**
After cloning the repo cd into it and run `python3 manage.py test` on the command line.
## **What are the relevant pivotal tracker stories?**
https://www.pivotaltracker.com/story/show/150627360

Commit:
feat(channels): Add team access test

-Ensure team members can access team data added by any member of the team

[Fixes #150627360]

Commit:
feat(channels): Add kinds test

-Ensure that bad kind of channel passed as an add argument cannot be added.

[Fixes #150627360]

Commit:
feat(channels): Add test for interference across team and non team members

Ensure that non team members cannot access team members channels and vice versa

[Fixes #150627360]

Commit:
feat(checks): Test team access only allowed to members

Ensure that team access on checks added is only allowed for team members, who can view checks added by members.

Team acces to non-team members is not allowed, and non-team members won't view it.

[Fixes #150627360]

Commit:
feat(pushover): Add test to ensure that the parameter priority is validated

Ensure that the parameter priority on pushover is passed correctly and validated to allow a response by the api

[Fixes #150627360]

Commit:
feat(pushover): Change the format of the function to align properly

Correct an error on alignment

[Fixes #150627360]

Commit:
feat(pushover): Change alignment of docstring on function

Correcting error on alignment

[Fixes #150627360]

Commit:
docs(channels): Add docstrings to module

Ensure that docstrings describe the module, class and functions

[Fixes #150627360]

Commit:
docs(checks): Add docstrings for checks

Ensure that docstrings describe the module, class and functions of testing adding checks

[Fixes #150627360]

Commit:
docs(pushover): Add docstrings, Improve variables

Add docstrings to module, functions and classes
Improve variable names
[Fixes, #150627360]

NgangaMaryanne commented on hc/front/tests/test_add_channel.py:
```diff
@@ -30,6 +36,7 @@ def test_it_trims_whitespace(self):
         self.assertEqual(q.count(), 1)
 
     def test_instructions_work(self):
+        """Test that instructions for each channel integration work while adding"""
```
double quotes detected. Previous comments use single quotes. Why the change?. Its good practice to stick to one convention.

NgangaMaryanne commented on hc/front/tests/test_add_pushbover.py:
```diff
@@ -1,40 +1,79 @@
+        1 -- High priority
+        2 -- Emergency priority
+        Parameter outside above range should return bad request status code.
+        """
```
⬆️

NgangaMaryanne commented on hc/front/tests/test_add_pushbover.py:
```diff
@@ -1,40 +1,79 @@
+        invalid_prio_param_2 = "pushover_user_key=a&nonce=n&prio=3"
+        r_with_invalid_prio = self.client.get("/integrations/add_pushover/?%s"
+                                              % invalid_prio_param_1)
+        assert r_with_invalid_prio.status_code == 400
```
other asserts use `self.assert`. When you decide on a particular convention, its good to stick with it. Please use `self.assert` for this.

Commit:
docs(fronttests): Change docstrings to single quotes

Used a convention of using single quotes on docstrings in stead of double quotes

[Fixes #150627360]

Commit:
feat(checks): Change assert method

Changed from using assert to using self.assert to stick to one convention

[Fixes #150627360]

Commit:
feat(pushover): Change assert method

Changed assert method to self. assert in stead of assert

[Fixes #150627360]

NgangaMaryanne commented:
LGTM
